### PR TITLE
Improve documentation.

### PR DESCRIPTION
**Pull request by rodrigogiraoserrao**
Multiple small fixes to documentation and docstrings.

### CODE REVIEW

File: src/textual/widgets/_directory_tree.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 @dataclass
 class DirEntry:
-    """Attaches directory information to a node."""
+    """Attaches directory information to a [DirectoryTree][textual.widgets.DirectoryTree] node."""
```
Comment (davep) [inline src/textual/widgets/_directory_tree.py:25]:
```suggestion
    """Attaches directory information to a [`DirectoryTree`][textual.widgets.DirectoryTree] node."""
```

File: src/textual/types.py
Comment (davep) [inline src/textual/types.py:1]: More of a general question for anyone reading: why do some things go into `types.py` but others not? I don't think I've ever seen a rationale for what goes where.

File: src/textual/widgets/_markdown.py [hunk 2]
```diff
@@ -1005,12 +1012,12 @@ def __init__(
 
     @property
     def document(self) -> Markdown:
-        """The Markdown document object."""
+        """The [Markdown][textual.widgets.Markdown] document widget."""
```
Comment (davep) [inline src/textual/widgets/_markdown.py:1015]:
```suggestion
        """The [`Markdown`][textual.widgets.Markdown] document widget."""
```

**Review by davep — APPROVED** [review]

File: src/textual/types.py
Comment (rodrigogiraoserrao) [reply src/textual/types.py:1]: Honestly, at this point I'm not sure.
Maybe we should make sure everything comes from its own original file?

I think in the beginning there were a couple of types that didn't really have a place they could call home, and so I exported them here.
Maybe we could clean this up a bit...
Do you think that's important now?

Comment (rodrigogiraoserrao) [reply src/textual/types.py:1]: I'll open a follow-up issue if you think that's a good idea.
(Thanks for the other two fixes, good catches!)

Comment (davep) [reply src/textual/types.py:1]: Just to confirm: no, not important now, and not anything that made sense to do in this PR; just a question for clarification for anyone who happened to be looking at this given the context, and possibly a catalyst for spinning off into another issue to consider this if there wasn't an obvious answer.

**Pull request merged**